plugins/module_utils/_api_data.py

```diff
@@ -3829,6 +3829,30 @@ def join_path(path):
             },
         ),
     ),
+    ('queue', 'simple'): APIData(
+        unversioned=VersionedAPIData(
+            primary_keys=('name', ),
+            fully_understood=True,
+            fields={
+                'bucket-size': KeyInfo(default='0.1'),
+                'burst-limit': KeyInfo(default=0),
+                'burst-threshold': KeyInfo(default=0),
+                'burst-time': KeyInfo(default='0s'),
+                'comment': KeyInfo(can_disable=True, remove_value=''),
+                'disabled': KeyInfo(default=False),
+                'limit-at': KeyInfo(default=0),
+                'max-limit': KeyInfo(default=0),
+                'name': KeyInfo(),
+                'packet-marks': KeyInfo(default=''),
+                'parent': KeyInfo(default='none'),
+                'priority': KeyInfo(default=8),
```

Comment:
Are you sure about those values? Unfortunately I cannot test it on my own right now via Ansible but manually set I get the error message `expected value of download-priority (line 1 column 39)`. The value `8/8` is the default for me. 

The following is the output of `/queue simple print detail` for me on RouterOS 7.13 where I only specified the `target` manually which you already correctly set as required.

`name="queue1" target=1.1.1.12/32 parent=none packet-marks="" priority=8/8 queue=default-small/default-small limit-at=0/0 max-limit=0/0 burst-limit=0/0 burst-threshold=0/0 burst-time=0s/0s bucket-size=0.1/0.1`.

Reply:
I took most of the default keys from the existing implementation of 'queue tree'.  These values work as expected in an environment running 7.13.4.

The returned data does include the '/'s but they do not appear to be required as a part of the data set.

```
--- before
+++ after
@@ -1,3 +1,21 @@
 {
-    "data": []
+    "data": [
+        {
+            ".id": "*1",
+            "bucket-size": "0.1/0.1",
+            "burst-limit": "0/0",
+            "burst-threshold": "0/0",
+            "burst-time": "0s/0s",
+            "comment": "Core: 172msa-dc1-core1:bonding1-vlan318 {HU-xxx-xxx} [1G]",
+            "disabled": false,
+            "limit-at": "0/0",
+            "max-limit": "0/1000000000",
+            "name": "queue-bonding1-vlan318",
+            "packet-marks": "",
+            "parent": "none",
+            "priority": "8/8",
+            "queue": "default-small/default-small",
+            "target": "bonding1-vlan318"
+        }
+    ]
 }
```

I'm happy to update and test the defaults to be include splitting the upload/download values if you like but it does work as-is.

Reply:
Mhh interesting fact that the API accepts it but CLI doesn't. I then guess there are also no changes when rolled out again with only one value defined.
It seems it is working and it is not documented by MikroTik so :+1:

Reply:
I just spun up a test and it shows changes for me, please see below.

```
     "data": [
         {
             ".id": "*7",
-            "bucket-size": "0.1/0.1",
-            "burst-limit": "0/0",
-            "burst-threshold": "0/0",
-            "burst-time": "0s/0s",
+            "bucket-size": "0.1",
+            "burst-limit": 0,
+            "burst-threshold": 0,
+            "burst-time": "0s",
             "disabled": true,
-            "limit-at": "0/0",
-            "max-limit": "7000000/2000000",
+            "limit-at": 0,
+            "max-limit": "7M/2M",
             "name": "Limit wifi speed",
             "packet-marks": "",
             "parent": "none",
-            "priority": "8/8",
+            "priority": 8,
             "queue": "pcq-upload-default/pcq-download-default",
             "target": "10.10.2.0/24"
```

The only data I have provided is the following.

```
  - disabled: True
    max-limit: 7M/2M
    name: Limit wifi speed
    queue: pcq-upload-default/pcq-download-default
    target: 10.10.2.0/24
```

Could you please recheck?

Reply:
Can you please try ```max-limit: 7000000/2000000```?  The API will accept k/M/G/T etc... but converts it to bits on  output.  This causes it to always trigger a change.

I created a filter that does this conversion, allowing me to still use the abbreviated format.

Reply:
Tbh. I could live with that but what I wanted to point out is the change from e.g. `"priority": "8/8"` to `"priority": 8` without manually defining it. So the defaults you set do lead to constant changes (at least for me). The diff you posted only shows a new entry. Would you mind applying this change, then do a `--check` run again and post the output, please?

Reply:
I've tested with your exact data with the results I've come to expect.

## Playbook
```yaml
- name: Test queue simple API
  hosts: mikrotik
  gather_facts: false
  module_defaults:
    community.routeros.api_modify:
      hostname: "{{ ansible_host }}"
      username: "{{ ansible_ssh_user }}"
      password: "{{ ansible_ssh_password }}"

  tasks:
    - name: Remove simple queues
      community.routeros.api_modify:
        path: queue simple
        handle_absent_entries: remove
        data: []

    - name: Add simple queue (Create)
      community.routeros.api_modify:
        path: queue simple
        data:
          - disabled: True
            max-limit: 7M/2M
            name: Limit wifi speed
            queue: pcq-upload-default/pcq-download-default
            target: 10.10.2.0/24

    - name: Add simple queue (Test for change with 7M/2M)
      community.routeros.api_modify:
        path: queue simple
        data:
          - disabled: True
            max-limit: 7M/2M
            name: Limit wifi speed
            queue: pcq-upload-default/pcq-download-default
            target: 10.10.2.0/24

    - name: Add simple queue (Test for change with 7000000/2000000)
      community.routeros.api_modify:
        path: queue simple
        data:
          - disabled: True
            max-limit: 7000000/2000000
            name: Limit wifi speed
            queue: pcq-upload-default/pcq-download-default
            target: 10.10.2.0/24
```

## With --check
```diff
$ ansible-playbook playbooks/queue_simple_test.yml -l gns3-lab-core1 --diff --check

PLAY [Test queue simple API] *********************************************************************************************************************************

TASK [Remove simple queues] **********************************************************************************************************************************
--- before
+++ after
@@ -1,20 +1,3 @@
 {
-    "data": [
-        {
-            ".id": "*9",
-            "bucket-size": "0.1/0.1",
-            "burst-limit": "0/0",
-            "burst-threshold": "0/0",
-            "burst-time": "0s/0s",
-            "disabled": true,
-            "limit-at": "0/0",
-            "max-limit": "7000000/2000000",
-            "name": "Limit wifi speed",
-            "packet-marks": "",
-            "parent": "none",
-            "priority": "8/8",
-            "queue": "pcq-upload-default/pcq-download-default",
-            "target": "10.10.2.0/24"
-        }
-    ]
+    "data": []
 }

changed: [gns3-lab-core1]

TASK [Add simple queue (Create)] *****************************************************************************************************************************
--- before
+++ after
@@ -8,7 +8,7 @@
             "burst-time": "0s/0s",
             "disabled": true,
             "limit-at": "0/0",
-            "max-limit": "7000000/2000000",
+            "max-limit": "7M/2M",
             "name": "Limit wifi speed",
             "packet-marks": "",
             "parent": "none",

changed: [gns3-lab-core1]

TASK [Add simple queue (Test for change with 7M/2M)] *********************************************************************************************************
--- before
+++ after
@@ -8,7 +8,7 @@
             "burst-time": "0s/0s",
             "disabled": true,
             "limit-at": "0/0",
-            "max-limit": "7000000/2000000",
+            "max-limit": "7M/2M",
             "name": "Limit wifi speed",
             "packet-marks": "",
             "parent": "none",

changed: [gns3-lab-core1]

TASK [Add simple queue (Test for change with 7000000/2000000)] ***********************************************************************************************
ok: [gns3-lab-core1]

PLAY RECAP ***************************************************************************************************************************************************
gns3-lab-core1             : ok=4    changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

## Without --check
```diff
$ ansible-playbook playbooks/queue_simple_test.yml -l gns3-lab-core1 --diff

PLAY [Test queue simple API] *********************************************************************************************************************************

TASK [Remove simple queues] **********************************************************************************************************************************
--- before
+++ after
@@ -1,20 +1,3 @@
 {
-    "data": [
-        {
-            ".id": "*9",
-            "bucket-size": "0.1/0.1",
-            "burst-limit": "0/0",
-            "burst-threshold": "0/0",
-            "burst-time": "0s/0s",
-            "disabled": true,
-            "limit-at": "0/0",
-            "max-limit": "7000000/2000000",
-            "name": "Limit wifi speed",
-            "packet-marks": "",
-            "parent": "none",
-            "priority": "8/8",
-            "queue": "pcq-upload-default/pcq-download-default",
-            "target": "10.10.2.0/24"
-        }
-    ]
+    "data": []
 }

changed: [gns3-lab-core1]

TASK [Add simple queue (Create)] *****************************************************************************************************************************
--- before
+++ after
@@ -1,3 +1,20 @@
 {
-    "data": []
+    "data": [
+        {
+            ".id": "*A",
+            "bucket-size": "0.1/0.1",
+            "burst-limit": "0/0",
+            "burst-threshold": "0/0",
+            "burst-time": "0s/0s",
+            "disabled": true,
+            "limit-at": "0/0",
+            "max-limit": "7000000/2000000",
+            "name": "Limit wifi speed",
+            "packet-marks": "",
+            "parent": "none",
+            "priority": "8/8",
+            "queue": "pcq-upload-default/pcq-download-default",
+            "target": "10.10.2.0/24"
+        }
+    ]
 }

changed: [gns3-lab-core1]

TASK [Add simple queue (Test for change with 7M/2M)] *********************************************************************************************************
changed: [gns3-lab-core1]

TASK [Add simple queue (Test for change with 7000000/2000000)] ***********************************************************************************************
ok: [gns3-lab-core1]

PLAY RECAP ***************************************************************************************************************************************************
gns3-lab-core1             : ok=4    changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

Reply:
Sorry it took me so long to respond. I tried it with version 7.13 and 7.13.4 but I still do get other results than you it seems. With your version I always get changes regardless of using `--check` or not. When using `--diff --check` I do get the differences shown but with only `--diff` I just see the following

```
PLAY [mikrotik_devices] ***********************************************************************************************

TASK [Manage /queue/simple] *******************************************************************************************
changed: [dev1.example.com]
changed: [dev2.example.com]
changed: [dev3.example.com]
changed: [dev4.example.com]
changed: [dev5.example.com]
changed: [dev6.example.com]
changed: [dev7.example.com]
changed: [dev8.example.com]

PLAY RECAP ************************************************************************************************************
dev3.example.com : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev4.example.com : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev5.example.com : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev1.example.com : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev2.example.com : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev7.example.com : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev6.example.com : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev8.example.com : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

So the API changes something but RouterOS transformed it so that when checking back for the result it is the same as before. That is not desirable IMHO.
When defining the defaults as MikroTik uses them (in `_api_data.py`) I get the following (IMHO desirable) result:

```
PLAY [mikrotik_devices] ***********************************************************************************************

TASK [Manage /queue/simple] *******************************************************************************************
ok: [dev1.example.com]
ok: [dev2.example.com]
ok: [dev3.example.com]
ok: [dev4.example.com]
ok: [dev5.example.com]
ok: [dev6.example.com]
ok: [dev7.example.com]
ok: [dev8.example.com]

PLAY RECAP ************************************************************************************************************
dev4.example.com : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev3.example.com : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev5.example.com : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev1.example.com : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev2.example.com : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev7.example.com : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev6.example.com : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
dev8.example.com : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```